spec/APNS.spec.js

```diff
@@ -1,68 +1,31 @@
-var APNS = require('../src/APNS');
+let APNS = require('../src/APNS');
 
 describe('APNS', () => {
 
-  it('can initialize with single cert', (done) => {
-    var args = {
-      cert: 'prodCert.pem',
-      key: 'prodKey.pem',
+  it('can initialize with cert', (done) => {
+    let args = {
+      cert: new Buffer('testCert'),
+      key: new Buffer('testKey'),
       production: true,
-      bundleId: 'bundleId'
+      topic: 'topic'
     }
-    var apns = new APNS(args);
+    let apns = new APNS(args);
 
-    expect(apns.conns.length).toBe(1);
-    var apnsConnection = apns.conns[0];
-    expect(apnsConnection.index).toBe(0);
-    expect(apnsConnection.bundleId).toBe(args.bundleId);
+    expect(apns.providers.length).toBe(1);
+    let apnsProvider = apns.providers[0];
+    expect(apnsProvider.index).toBe(0);
+    expect(apnsProvider.topic).toBe(args.topic);
     // TODO: Remove this checking onec we inject APNS
-    var prodApnsOptions = apnsConnection.options;
+    let prodApnsOptions = apnsProvider.client.config;
     expect(prodApnsOptions.cert).toBe(args.cert);
     expect(prodApnsOptions.key).toBe(args.key);
     expect(prodApnsOptions.production).toBe(args.production);
     done();
   });
 
-  it('can initialize with multiple certs', (done) => {
```

Comment:
this is supported for different topics/bundleIds.
But I haven't found any possibility, to implement a retry for notifications, that a rejected by apns because of wrong provider or wrong production- / test-device.

Example: You have two provider, one for production use and one for development. Both share the same topic/bundleId. So `_chooseProviders` would return `[prodProvider, devProvider]` based on priority-sorting in constructor. So `send`-Method uses the first one to send the notification. If device is registered using apple-development-provisioning-profile, the notification would not be delivered, because of wrong provider. In parse-device-object is no flag, that describes, wether this device is using development- or production-profile.

In "old" parse-push-adapter (using apn version 1.X), there was a retry-functionality. This isn't available in my version.

I will add this test-case again to test different topics.